src/OpenTelemetry.AutoInstrumentation.Native/continuous_profiler.h

```diff
@@ -35,6 +35,64 @@ extern "C"
     // ReSharper restore CppInconsistentNaming
 }
 
+namespace continuous_profiler
+{
+struct FunctionIdentifier
+{
+    mdToken  function_token;
+    ModuleID module_id;
+    bool     is_valid;
+
+    bool operator==(const FunctionIdentifier& p) const
+    {
+        return function_token == p.function_token && module_id == p.module_id && is_valid == p.is_valid;
+    }
+};
+
+struct FunctionIdentifierResolveArgs
+{
+    FunctionID  function_id;
+    COR_PRF_FRAME_INFO frame_info;
+
+    bool operator==(const FunctionIdentifierResolveArgs& p) const
+    {
+        return function_id == p.function_id && frame_info == p.frame_info;
+    }
+};
+}
+
+template <>
+struct std::hash<continuous_profiler::FunctionIdentifier>
+{
+    std::size_t operator()(const continuous_profiler::FunctionIdentifier& k) const noexcept
+    {
+        using std::hash;
+        using std::size_t;
+        using std::string;
+
+        const std::size_t h1 = std::hash<mdToken>()(k.function_token);
+        const std::size_t h2 = std::hash<ModuleID>()(k.module_id);
+
```

Comment:
We also need to include bool is_valid flag. Two equal objects should produce same hash. If two objects with same token and module ID but different bool member will produce same hash.